boa/src/syntax/parser/statement/try_stm/mod.rs

```diff
@@ -63,38 +69,24 @@ impl TokenParser for TryStatement {
             ));
         }
 
-        // CATCH
         let (catch, param) = if next_token.kind == TokenKind::Keyword(Keyword::Catch) {
-            // Catch binding
-            cursor.expect(Punctuator::OpenParen, "catch in try statement")?;
-            // TODO: should accept BindingPattern
-            let tok = cursor.next().ok_or(ParseError::AbruptEnd)?;
-            let catch_param = if let TokenKind::Identifier(s) = &tok.kind {
-                Node::local(s)
-            } else {
-                return Err(ParseError::Expected(
-                    vec![TokenKind::identifier("identifier")],
-                    tok.clone(),
-                    "catch in try statement",
-                ));
-            };
-            cursor.expect(Punctuator::CloseParen, "catch in try statement")?;
-
-            // Catch block
-            (
-                Some(
-                    Block::new(self.allow_yield, self.allow_await, self.allow_return)
-                        .parse(cursor)?,
-                ),
-                Some(catch_param),
-            )
+            match Catch::new(self.allow_yield, self.allow_await, self.allow_return).parse(cursor) {
+                Ok((catch, param)) => (catch, param),
+                Err(e) => return Err(e),
+            }
         } else {
             (None, None)
         };
 
-        // FINALLY
-        let finally_block = if cursor.next_if(Keyword::Finally).is_some() {
-            Some(Block::new(self.allow_yield, self.allow_await, self.allow_return).parse(cursor)?)
+        let next_token = cursor.peek(0);
+        let finally_block = if next_token.is_some()
+            && next_token.unwrap().kind == TokenKind::Keyword(Keyword::Finally)
```

Comment:
I would use a match statement to avoid `unwrap()` should be more efficient, too.

Reply:
Done